src/Error/ExceptionTrap.php

```diff
@@ -334,7 +335,6 @@ public function logException(Throwable $exception, ?ServerRequestInterface $requ
     {
         $logger = $this->logger();
         $logger->log($exception, $request);
-        $this->dispatchEvent('Exception.beforeRender', ['exception' => $exception]);
```

Comment:
This was an awkward place before. But this ended up here to support the logic needed in `ErrorHandlerMiddleware` with as few public methods a possible. There is clearly a test missing for this as the build passed.

My question is will users expect this event to fire from exceptions caught in the middleware? What about internal errors caught in the middleware? Originally my thought was 'of course they should work the same as globally caught exceptions'. If folks are ok with not supporting that usage, then we're safe to remove this. Otherwise we need another public method and to call that method alongside `$renderer->render()`.

Reply:
In that case, we probably can't call it beforeRender() or need to revert my last PR which wraps logException().

Reply:
What about adding a `render()` method to `ExceptionTrap` that includes:

* Triggering beforeRender
* Getting the renderer and rendering the response.

Reply:
That seems ok. I haven't had time to look into why logException was like that.